### PR TITLE
Freon but I made it cooler [READY]

### PR DESCRIPTION
**Pull request by yoyobatty**
[Guidelines]: # (Be sure that your PR follows our guidelines, such as modularization and comment standards. You can read more about the subject here: https://github.com/HippieStation/HippieStation/blob/master/hippiestation/README.md )

## Changelog
:cl: YoYoBatty
add: DuPont has re-added their brand of refrigerants known as Freon to the station. 
/:cl:

## About The Pull Request
Title. In order to react, it removes nitrogen from the air and air temp must be 0C or below. It gets very very cold if there are sufficient amounts of nitrogen moles in the air.

## Why It's Good For The Game
Old hippie feature that has some nostalgia to it.

### CODE REVIEW

File: hippiestation/code/modules/atmospherics/gasmixtures/reactions.dm [hunk 1]
```diff
@@ -46,8 +45,27 @@
...
+//freon: does a freezy thing?
+/datum/gas_reaction/freon
+	priority = 1
+	name = "Freon"
+	id = "freon"
 
-		return FALSE
+/datum/gas_reaction/freon/init_reqs()
+	min_requirements = list(/datum/gas/freon = MOLES_GAS_VISIBLE,
+							/datum/gas/oxygen = MINIMUM_MOLE_COUNT,
```
Comment (yoyobatty) [inline hippiestation/code/modules/atmospherics/gasmixtures/reactions.dm]: I guess I should take out the o2 requirements

Comment (WarRage333) [reply hippiestation/code/modules/atmospherics/gasmixtures/reactions.dm]: I think you could bough it at cargo before too right?

**Comment by WarRage333** [comment]
I think you could buy this at cargo before too right?

**Comment by yoyobatty** [comment]
Believe so, it's a big work in progress and the code to make Freon work is still in game. I even made it so when it colours stuff like lights, they don't completely black them out, which is nice.

I am hopefully going to make this more complex to allow use in heat exchange systems similar to how AC systems work... But that bridge will be crossed once I get there

**Comment by SireTurret** [comment]
Y'know, if I tried to bring back freon myself it'd probably take ages. Thanks, Yoyo

**Comment by Manly-Dwarf** [comment]
Would not remind the return of Freon as long as it's not available to Atmos immediately.
Might be cool to see some of the techs figure out how to fill up canisters with freon and then freeze the entire station.

**Comment by yoyobatty** [comment]
I probably could allow Freon to be made by supercooling a combination of various gases, such as tritium, co2 and no2 for examples. Production of it probably could produce a ton of heat so that you can't simply spam it super easy. It's a powerful gas that can cause a lot of grief or save the supermatter, it may as well be a challenge to produce.

**Comment by Majick1** [comment]
+1 no reason not to have it really

**Review by ghost — DISMISSED** [review]

**Comment by Toxici11i** [comment]
yes pls +1

**Comment by yoyobatty** [comment]
I am due to make a reaction to allow creation of Freon. I also need to make it interact with other gases more appropriately. Thanks to the saviors at TG, any type of gas you can make at any point is auto added to air alarms and atmospheric machinery so I didn't need to go out of my way to program that it.

**Comment by yoyobatty** [comment]
Still working on this, adding a production reaction.

**Comment by Manly-Dwarf** [comment]
AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA

**Comment by SireTurret** [comment]
AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA

**Comment by yoyobatty** [comment]
Remind me to finish this

**Comment by ghost** [comment]
finish it @yoyobatty

**Comment by yoyobatty** [comment]
Ready.

**Comment by Manly-Dwarf** [comment]
AAAAAAAAAAAAAAAAAAAAAAA

**Pull request merged**